certification/code/certificates.py

```diff
@@ -0,0 +1,70 @@
+#!/usr/bin/env python
+
+'''
+Generate a certificate from a template.
+
+* Requires the python package 'cairosvg' to be installed.
+  Please visit http://cairosvg.org/ for install instructions.
+* Some systems may also need to have 'cairo' installed.
+  Please visit http://cairographics.org/download/ for the same.
+'''
+
+import sys
+import os
+import re
+import tempfile
+import cairosvg
+from django.conf import settings
+
+#---------------------------------------------------------------
+# Main Functions
+
+def generate(args):
+    '''Process a single entry and returns their certificate'''
+    root_dir = os.path.dirname(__file__)
+    template_path = construct_template_path(root_dir, args.badge_type)
+    return create_certificate(template_path, args.params, args.cert_id)
+
+
+def create_certificate(template_path, params, cert_id):
+    '''Creates and returns a single certificate.'''
+
+    with open(template_path, 'r') as reader:
+        template = reader.read()
+    check_template(template, params)
+
+    for key, value in params.items():
+        pattern = '{{' + key + '}}'
+        template = template.replace(pattern, value)
```

Comment:
That's really a nice thought, but I think that should be avoided for now, mainly because it's deeply integrated with the workshops app. We're only generating certificates for the workshops app, plus it relies on a model which is heavily dependent on the workshops app.